lib/utils/utf8.c

```diff
@@ -79,3 +87,39 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const unsigned char *str, size_t maxlen)
+{
+	size_t i = 0, nbyte = 0;
+	size_t len = strnlen(str, maxlen);
+
+	/* It will also return false */
+	if (str == NULL) {
+		return false;
+	}
```

Comment:
The `NULL` check needs to be done before `strnlen`

Reply:
Oh, thank you for reminding me. I almost forgot.